ERCS/erc-7540.md

```diff
@@ -224,30 +310,107 @@ MUST be emitted when a redemption Request is submitted using the `requestRedeem`
   type: event
 
   inputs:
-    - name: sender
-      indexed: true
-      type: address
-    - name: operator
+    - name: receiver
       indexed: true
       type: address
     - name: owner
       indexed: true
       type: address
+    - name: rid
+      indexed: true
+      type: uint256
+    - name: sender
+      indexed: false
+      type: uint256
     - name: assets
       indexed: false
       type: uint256
 ```
 
+### Request Callbacks
+
+All methods which initiate a request (including `rid==0`) include a `data` parameter, which if nonzero length MUST send a callback to the receiver. 
+
+There are two interfaces, `ERC7540DepositReceiver` and `ERC7540RedeemReceiver` which each define the single callback method to be called.
+
+#### ERC7540DepositReceiver
+The interface to be called on `requestDeposit`.
+
+`operator` is the `msg.sender` of the original `requestDeposit` call. `owner` is the `owner` of the `requestDeposit`. `rid` is the output `rid` of the `requestDeposit` and `data` is the `data` of the `requestDeposit`.
+
+This function MUST return `TODO` upon successful execution of the callback.
+
+```yaml
+- name: onERC7540DepositReceived
+  type: function
+
+  inputs:
+    - name: operator
+      type: address
+    - name: owner
+      type: address
+    - name: rid
+      type: uint256
+    - name: data
+      type: bytes
+  outputs:
+    - name: interfaceId
+      type: bytes4
+```
+
+
+#### ERC7540RedeemReceiver
+The interface to be called on `requestRedeem`.
+
+`operator` is the `msg.sender` of the original `requestRedeem` call. `owner` is the `owner` of the `requestRedeem`. `rid` is the output `rid` of the `requestRedeem` and `data` is the `data` of the `requestRedeem`.
+
+This function MUST return `TODO` upon successful execution of the callback.
+
+```yaml
+- name: onERC7540RedeemReceived
+  type: function
+
+  inputs:
+    - name: operator
+      type: address
+    - name: owner
+      type: address
+    - name: rid
+      type: uint256
+    - name: data
+      type: bytes
+  outputs:
+    - name: interfaceId
+      type: bytes4
+```
+
 ### [ERC-165](./eip-165.md) support
 
 Smart contracts implementing this standard MUST implement the [ERC-165](./eip-165.md) `supportsInterface` function.
 
-Asynchronous deposit Vaults MUST return the constant value `true` if `0xea446681` is passed through the `interfaceID` argument.
+Asynchronous deposit Vaults MUST return the constant value `true` if TODO `0xea446681` is passed through the `interfaceID` argument.
+
+Asynchronous redemption Vaults MUST return the constant value `true` if TODO `0x2e9dd5bd` is passed through the `interfaceID` argument.
 
-Asynchronous redemption Vaults MUST return the constant value `true` if `0x2e9dd5bd` is passed through the `interfaceID` argument.
+ERC7540DepositReceiver contracts MUST return the constant value `true` if TODO is passed through the `interfaceID` argument.
```

Comment:
Do the receiver contracts need to implement ERC-165? Just ensuring that they return `bytes4(keccak256("onERC7540DepositReceived(address,address,uint256,bytes)"))` on `onERC7540DepositReceived` calls, and vice verse for redeem, is sufficient, I think?

Reply:
You're right, lets leave it out and I'll reword